tdms/src/iterator.cpp

```diff
@@ -871,10 +871,10 @@ void execute_simulation(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs
 
   //work out if we have any disperive materials
   bool is_disp = is_dispersive(materials, gamma, params.dt, I_tot, J_tot, K_tot);
-  //work out if we have conductive background
-  bool is_cond = is_conductive(rho_cond, I_tot, J_tot, K_tot);
-  //work out if we have a dispersive background
-  if (params.is_disp_ml) params.is_disp_ml = is_dispersive_ml(ml, K_tot);
+  //work out if we have conductive background: background is conductive if at least one entry exceeds 1e-15
+  bool is_cond = !(rho_cond.all_elements_less_than(1e-15, I_tot + 1, J_tot + 1, K_tot + 1));
+  // work out if we have a dispersive background
+  if (params.is_disp_ml) params.is_disp_ml = ml.is_dispersive(K_tot);

```

Comment:
`ml` is the "matched" layer? So somewhere there's a `DispersiveLayer ml` declared?

Is it a pain to rename `ml` to `matched_layer`? Does it make sense?

Reply:
[Right here](https://github.com/UCL/TDMS/blob/wgraham-refactor_iterator_3/tdms/src/iterator.cpp#L438) approx 400 lines ago.

Have run a VSCode refactor over the ml symbol